includes/Modules/Analytics.php

```diff
@@ -937,115 +937,107 @@ protected function parse_data_response( $method, $datapoint, $response ) {
-					$response = array(
-						// TODO: Parse this response to a regular array.
-						'properties' => $response->getItems(),
+					/* @var \Google_Service_Analytics_Webproperties $response listManagementWebproperties response. */
+					$properties = (array) $response->getItems();
+					$response   = array(
+						'properties' => $properties,
 						'profiles'   => array(),
 					);
-					if ( 0 === count( $response['properties'] ) ) {
-						return new WP_Error( 'google_analytics_properties_empty', __( 'No Google Analytics properties found. Please go to Google Anlytics to set one up.', 'google-site-kit' ), array( 'status' => 500 ) );
+
+					if ( 0 === count( $properties ) ) {
+						return new WP_Error(
+							'google_analytics_properties_empty',
+							__( 'No Google Analytics properties found. Please go to Google Analytics to set one up.', 'google-site-kit' ),
+							array( 'status' => 500 )
+						);
 					}
-					$found_account_id  = false;
-					$found_property_id = false;
-					$property_id       = $this->get_data( 'property-id' );
-					if ( ! is_wp_error( $property_id ) ) {
-						foreach ( $response['properties'] as $property ) {
-							if ( $property->getId() === $property_id ) {
-								$found_account_id  = $property->getAccountId();
-								$found_property_id = $property->getId();
-								break;
-							}
+
+					$property_id    = $this->get_data( 'property-id' );
+					$found_property = new \Google_Service_Analytics_Webproperty();
+					$current_url    = untrailingslashit( $this->context->get_reference_site_url() );
+					$current_urls   = $this->permute_site_url( $current_url );
+
+					// If there's no match for the saved account ID, try to find a match using the properties of each account.
+					foreach ( $properties as $property ) {
+						/* @var \Google_Service_Analytics_Webproperty $property Property instance. */
+						if (
+							// Attempt to match by property ID.
+							$property->getId() === $property_id ||
+							// Attempt to match by site URL, with and without http/https and 'www' subdomain.
+							in_array( untrailingslashit( $property->getWebsiteUrl() ), $current_urls, true )
+						) {
+							$found_property              = $property;
+							$response['matchedProperty'] = $property;
+							break;
 						}
 					}
-					if ( empty( $found_account_id ) || empty( $found_property_id ) ) {
-						$found_account_id  = $response['properties'][0]->getAccountId();
-						$found_property_id = $response['properties'][0]->getId();
+
+					// If no match is found, fetch profiles for the first property if available.
+					if ( ! $found_property->getAccountId() && $properties ) {
+						$found_property = array_shift( $properties );
 					}
+
 					$profiles = $this->get_data(
 						'profiles',
 						array(
-							'accountId'  => $found_account_id,
-							'propertyId' => $found_property_id,
+							'accountId'  => $found_property->getAccountId(),
+							'propertyId' => $found_property->getId(),
```

Comment:
If `$found_property` is the "empty" property from above, we probably shouldn't even call `profiles` because it will definitely be empty then.